### DISCUSSION TITLE
websocket-async random receive while continuously sending

### DISCUSSION THREAD

**Discussion by qqjyh** [miguelgrinberg/microdot#258] [Q&A]
[Microdot-websocket-async] is great, not only for its superior performance, but also for its ease of programming. Could please add some in the example ws_ sync. py of [microdot-main\examples\websocket] that allows for random reception while continuously sending (instead of waiting for reception before sending) ?

**Comment by qqjyh** [comment]
[await ws.receive()] How to eliminate the preceding await ?

**Reply by miguelgrinberg** [reply]
You have to start a separate task to receive. I'll add an example when I find a bit of time, but it would look something like this:

```python
@app.route('/echo')
@with_websocket
async def websocket(request, ws):
    async def receive():
        while True:
            data = await ws.receive()
            print('received', data)

    asyncio.create_task(receive())
    while True:
        await ws.send(...)
        await asyncio.sleep(1)
```

**Reply by qqjyh** [reply]
Thank you very much to the master for providing me with critical solutions in a busy work environment, especially the program examples given, which have inspired a new asynchronous programming technique for everyone.

**Reply by qqjyh** [reply]
Several weeks failed in finding mPython's websocket-client (running in the same environment as microdot-websocket) , so I can only ask the master to send a link at a convenient time ?

**Reply by miguelgrinberg** [reply]
A websocket client has nothing in common with microdot, so there is really nothing to demonstrate, you can just use your client on a separate task and that should be it.
